sdk/synapse/Azure.Analytics.Synapse.Spark/src/Customization/SparkSessionOperation.cs

```diff
@@ -157,5 +97,32 @@ private static bool IsJobComplete(string jobState, string livyState)
 
             return false;
         }
+
+        async Task<Response<SparkSession>> IOperation<SparkSession, SparkSession>.GetResponseAsync(CancellationToken cancellationToken) =>
+            await _client.RestClient.GetSparkSessionAsync(_value.Id, true, cancellationToken).ConfigureAwait(false);
+
+        Response<SparkSession> IOperation<SparkSession, SparkSession>.GetResponse(CancellationToken cancellationToken) =>
+            _client.RestClient.GetSparkSession(_value.Id, true, cancellationToken);
+
+        OperationState<SparkSession> IOperation<SparkSession, SparkSession>.UpdateState(Response<SparkSession> response)
+        {
+            var state = new OperationState<SparkSession>();
+
+            if (IsJobComplete(response.Value.Result.ToString(), response.Value.State))
+            {
+                if (StringComparer.OrdinalIgnoreCase.Equals("error", response?.Value?.State))
+                {
+                    state.Succeeded = false;
+                    state.OperationFailedException = new RequestFailedException("SparkBatchOperation ended in state error");
```

Comment:
Do failed operations follow some common response schema like the failed requests? Can we have a reasonable common implementation of `OperationFailedException`?

Reply:
From my understanding, these are usually returned as 2xx (request succeeded, but operation failed), so there's not much we can infer from it without looking into the response. The flags used to communicate an error occurred differ across services.